velox/functions/prestosql/window/Rank.cpp

```diff
@@ -75,10 +111,23 @@ void registerRank(const std::string& name) {
       [name](
           const std::vector<TypePtr>& /*argTypes*/,
           const std::vector<column_index_t>& /*argIndices*/,
-          const TypePtr& /*resultType*/,
+          const TypePtr& resultType,
           velox::memory::MemoryPool* /*pool*/)
           -> std::unique_ptr<exec::WindowFunction> {
-        return std::make_unique<RankFunction>();
+        return std::make_unique<RankFunction<type>>(resultType);
       });
 }
+
+void registerRank(const std::string& name, const std::string& returnType) {
```

Comment:
The caller needn't pass returnType as a parameter here. Since rank, denserank are always bigint and percentrank always double, they can be hard-coded from here into the registerRankInternal calls.

Reply:
Got it, thanks!